pulser-pasqal/pulser_pasqal/pasqal_cloud.py

```diff
@@ -86,14 +84,26 @@ def create_batch(
                 "must be a real device, instance of 'Device'."
             )
 
-        self._sdk_connection.create_batch(
-            serialized_sequence=seq.serialize(),
-            jobs=jobs,
+        for params in jobs:
+            seq.build(**params.parameters)  # type: ignore
+
+        return self._sdk_connection.create_batch(
+            serialized_sequence=self._serialize_seq(
+                seq=seq, device_type=device_type
+            ),
+            jobs=[j.parameters for j in jobs],
```

Comment:
Following my suggestion for `JobParameters`, getting the `parameters` can fit nicely as a property or method